File: AutoIt.Foundation/AutoIt.Foundation.Common/LangAnaly/Model/Info/TokenInfo.cs
namespace AutoIt.Foundation.Common.LangAnaly
{
    public class TokenInfo: SymbolInfoBase
    {
        public TokenInfoState State { get; set; }

        public TokenInfo(TokenInfoState state, Symbol symbol, string value,int index,
            int line,int col):base(symbol,value,index,line,col)
        {
            this.State = state;
        }
    }
}

File: AutoIt.Foundation/AutoIt.Foundation.Web.Scripts/CodeEdit/LangAnaly/Model/TokenInfo.ts
module CodeEdit.LangAnaly.Model {
    //¼ÇºÅ
    export class TokenInfo extends CodeEdit.LangAnaly.Model.SymbolInfoBase {
        //×´Ì¬
        State: CodeEdit.LangAnaly.Model.TokenInfoState;

        static NullToken():TokenInfo {
            return new Model.TokenInfo(Model.TokenInfoState.Accept,
                null,
                null,
                -1,
                -1,
                -1);
        }

        constructor(state: CodeEdit.LangAnaly.Model.TokenInfoState,
            symbol: CodeEdit.LangAnaly.Model.Symbol,
            value: string,
            index: number,
            line: number,
            col: number) {
            super(symbol, value, line, col, index);
            this.State = state;
        }
    }
}